ash/src/device.rs

```diff
@@ -19,16 +19,24 @@ pub struct Device {
 
 impl Device {
     pub unsafe fn load(instance_fn: &vk::InstanceFnV1_0, device: vk::Device) -> Self {
-        let load_fn = |name: &std::ffi::CStr| {
-            mem::transmute((instance_fn.get_device_proc_addr)(device, name.as_ptr()))
-        };
+        Self::load_with(
+            |name: &std::ffi::CStr| {
+                mem::transmute((instance_fn.get_device_proc_addr)(device, name.as_ptr()))
+            },
+            device,
+        )
+    }
 
+    pub unsafe fn load_with(
+        mut load_fn: impl FnMut(&core::ffi::CStr) -> *const c_void,
```

Comment:
Making this `&mut dyn FnMut` could save downstream monomorphization on this cold path. Probably a marginal difference and slightly worse ergonomics, though.

Reply:
I think it'd have to be comparative benchmarking of code size and compile time.

In any case, we could construct a `&dyn mut FnMut` internally without breaking the interface introduced in this PR, thereby only having to monomorphize a trivial amount of code downstream. I'm unsure if that would actually work so long as the inner table `load`s are still generic (depends how smart rustc is), but we can stick with the nice ergonomic interface here regardless.